app/scheduling/directives/termCalendar.js

```diff
@@ -104,34 +104,104 @@ schedulingApp.directive("termCalendar", this.termCalendar = function ($rootScope
+				var r = (num >> 16) + amt;
+
+				if (r > 255) {
+					r = 255;
+				} else if (r < 0) {
+					r = 0;
+				}
+
+				var b = ((num >> 8) & 0x00FF) + amt;
+
+				if (b > 255) {
+					b = 255;
+				} else if (b < 0) { b = 0; }
+
+				var g = (num & 0x0000FF) + amt;
+
+				if (g > 255) {
+					g = 255;
+				} else if (g < 0) {
+					g = 0;
+				}
+
+				return (usePound ? "#" : "") + (g | (b << 8) | (r << 16)).toString(16);
+			};
+
 			var getActivities = function () {
 				// Each of these If blocks will add to a 'events array'
 				// The event making function will color them appropriately
 				var calendarActivities = [];
 
 				// Add Selected sectionGroup activities
 				if (scope.view.state.uiState.selectedSectionGroupId) {
-					var unstyledEvents = sectionGroupToActivityEvents(scope.view.state.sectionGroups.list[scope.view.state.uiState.selectedSectionGroupId]);
+					var sectionGroup = scope.view.state.sectionGroups.list[scope.view.state.uiState.selectedSectionGroupId];
+					var unstyledEvents = sectionGroupToActivityEvents(sectionGroup);
+					var tagColor = calculateTagColor(sectionGroup);
+
+					var textColor = tagColor ? "#FFFFFF" : activeEventTextColor;
+					var borderColor = tagColor ? tagColor : activeEventBorderColor;
+					var backgroundColor = tagColor ? tagColor : activeEventBackgroundColor;
+
 					calendarActivities = calendarActivities.concat(
-						styleCalendarEvents(unstyledEvents, activeEventBackgroundColor, activeEventBorderColor, activeEventTextColor)
+						styleCalendarEvents(unstyledEvents, backgroundColor, borderColor, textColor, tagColor)
 					);
 				}
 
 				// Add checked sectionGroups activities
 				if (scope.view.state.uiState.checkedSectionGroupIds.length > 0) {
 					scope.view.state.uiState.checkedSectionGroupIds.forEach(function (sgId) {
 						if (sgId !== scope.view.state.uiState.selectedSectionGroupId) {
+							var sectionGroup = scope.view.state.sectionGroups.list[sgId];
+							var course = scope.view.state.courses.list[sectionGroup.courseId];
 							var unstyledEvents = sectionGroupToActivityEvents(scope.view.state.sectionGroups.list[sgId]);
-							calendarActivities = calendarActivities.concat(
-								styleCalendarEvents(unstyledEvents)
-							);
+
+
+							var tagColor = calculateTagColor(sectionGroup);
+
+							if (tagColor) {
+								calendarActivities = calendarActivities.concat(
+									styleCalendarEvents(unstyledEvents, tagColor, tagColor, "white")
+								);
+							} else {
+								calendarActivities = calendarActivities.concat(
+									styleCalendarEvents(unstyledEvents)
+								);
+							}
 						}
 					});
 				}
 
 				return calendarActivities;
 			};
 
+			var calculateTagColor = function (sectionGroup) {
+				var tagColor = null;
+				var course = scope.view.state.courses.list[sectionGroup.courseId];
+
+				if (course.tagIds.length > 0) {
+					var tag = scope.view.state.tags.list[course.tagIds[0]];
+					if (tag.color) {
+						tagColor = tag.color;
```

Comment:
Do we want to return early once tagColor is set? Or are we taking the last tag color?